src/controller/model/endpoint.ts

```diff
@@ -261,26 +261,36 @@ class Endpoint extends Entity {
         return null;
     }
 
-    public sendPendingRequests(): void {
-        [...this.pendingRequests].forEach(async (r) => {
-            this.pendingRequests.splice(this.pendingRequests.indexOf(r), 1);
+    public async sendPendingRequests(): Promise<void> {
+        while (this.pendingRequests.length > 0) {
+            const r = this.pendingRequests.shift();
             try {
                 const result = await r.func();
                 r.resolve(result);
             } catch (error) {
                 r.reject(error);
             }
-        });
+        }
     }
 
-    public async sendRequest<Type>(func: () => Promise<Type>, sendWhenActive: boolean): Promise<Type> {
-        if (sendWhenActive) {
-            return new Promise((resolve, reject): void =>  {
-                this.pendingRequests.push({func, resolve, reject});
-            });
-        } else {
-            return func();
+    private async sendRequest<Type>(func: () => Promise<Type>, sendWhenActive: boolean): Promise<Type> {
+        if (this.pendingRequests.length === 0) {
```

Comment:
Why is the request only send when pendingRequests are empty? I would expect that a request with `sendWhenActive = false` would always be send directly.

Reply:
Hmm, true. I'll change that.

The idea was that if things had started queuing, we'd keep queuing for some amount of ordering, but it does behave a bit odd of you're asking for no queueing.